packages/enzyme-test-suite/test/ShallowWrapper-spec.jsx

```diff
@@ -666,6 +666,48 @@ describe('shallow', () => {
           expect(consumer.text()).to.equal('foo');
         });
       });
+
+      describe('shallow() on Provider and Consumer through .contextType', () => {
+
+        const { Provider } = React.createContext('howdy!');
+
+        class OuterComponent extends React.Component {
+          render() {
+            return (
+              <Provider value="foo"><InnerComponent /></Provider>
+            );
+          }
+        }
+
+        class WrappingComponent extends React.Component {
+          render() {
+            const { children } = this.props;
+            return (
+              <Provider value="foo">{ children }</Provider>
+            );
+          }
+        }
+
+        class InnerComponent extends React.Component {
+          render() {
+            return this.context;
+          }
+        }
+
+        InnerComponent.contextType = Provider;
+
+        it('works on a Provider', () => {
+          const wrapper = shallow(<OuterComponent />);
+          const provides = wrapper.find(Provider).dive();
+          const provider = provides.find(InnerComponent).shallow();
```

Comment:
```suggestion
          const provider = provides.find(InnerComponent).dive();
```
any reason this wouldn't work?